hal_st/stm32fxxx/UartStm.cpp

```diff
@@ -62,7 +69,7 @@ namespace hal
         uartHandle.Init.WordLength = config.parity == USART_PARITY_NONE ? USART_WORDLENGTH_8B : USART_WORDLENGTH_9B;
         uartHandle.Init.StopBits = USART_STOPBITS_1;
         uartHandle.Init.Parity = config.parity;
-        uartHandle.Init.Mode = USART_MODE_TX_RX;
+        uartHandle.Init.Mode = USART_MODE_TX;
```

Comment:
Are you then enabling the RX mode with CR1_RE?

Reply:
correct, `USART_MODE_x_y` is defined as:
```
USART_MODE_TX = USART_CR1_TE
USART_MODE_RX = USART_CR1_RE
USART_MODE_TX_RX = USART_CR1_TE | USART_CR1_RE
```